apps/meteor/client/views/admin/moderation/UserReports/UserReportInfo.tsx

```diff
@@ -0,0 +1,135 @@
+import type { IUser, UserReport, Serialized } from '@rocket.chat/core-typings';
+import {
+	Box,
+	Callout,
+	StatesAction,
+	StatesActions,
+	StatesIcon,
+	StatesTitle,
+	ContextualbarFooter,
+	FieldGroup,
+	Field,
+	FieldLabel,
+	FieldRow,
+	ContextualbarSkeleton,
+} from '@rocket.chat/fuselage';
+import { useEndpoint, useTranslation } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+import React, { useMemo } from 'react';
+
+import { ContextualbarScrollableContent } from '../../../../components/Contextualbar';
+import GenericNoResults from '../../../../components/GenericNoResults';
+import UserCard from '../../../../components/UserCard';
+import { useFormatDate } from '../../../../hooks/useFormatDate';
+import ReportReason from '../helpers/ReportReason';
+import UserColumn from '../helpers/UserColumn';
+import UserContextFooter from './UserContextFooter';
+
+const UserReportInfo = ({ userId }: { userId: string }) => {
+	const t = useTranslation();
+	const getUserReports = useEndpoint('GET', '/v1/moderation.user.reportsByUserId');
+	const formatDateAndTime = useFormatDate();
+
+	const {
+		data: report,
+		refetch: reloadUsersReports,
+		isLoading,
+		isSuccess,
+		isError,
+		dataUpdatedAt,
+	} = useQuery(['moderation.usersReport', userId], async () => getUserReports({ userId }));
+
+	const userProfile = useMemo(() => {
+		if (!report?.user) {
+			return null;
+		}
+
+		const { username, name } = report.user;
+		return <UserColumn key={dataUpdatedAt} username={username} name={name} isProfile={true} />;
+	}, [report?.user, dataUpdatedAt]);
+
+	const userEmails = useMemo(() => {
+		if (!report?.user?.emails) {
+			return [];
+		}
+		return report.user.emails.map((email) => email.address);
+	}, [report]);
+
+	const renderUserDetails = (user: Serialized<IUser>) => {
+		return (
+			<Box>
+				<FieldGroup>
+					<Field>{userProfile}</Field>
+					<Field>
+						<FieldLabel>{t('Roles')}</FieldLabel>
+						<FieldRow justifyContent='flex-start' spacing={1}>
+							{user.roles.map((role, index) => (
+								<UserCard.Role key={index}>{role}</UserCard.Role>
+							))}
+						</FieldRow>
+					</Field>
+					<Field>
+						<FieldLabel>{t('Email')}</FieldLabel>
+						<FieldRow>{userEmails.join(', ')}</FieldRow>
+					</Field>
+					<Field>
+						<FieldLabel>{t('Created_at')}</FieldLabel>
+						<FieldRow>{formatDateAndTime(user.createdAt)}</FieldRow>
+					</Field>
+				</FieldGroup>
+			</Box>
+		);
+	};
+
+	const renderDeletedUserWarning = () => {
+		return (
+			<Box>
+				<Callout mbs={8} type='warning' icon='warning'>
+					{t('Moderation_User_deleted_warning')}
+				</Callout>
+			</Box>
+		);
+	};
```

Comment:
Sorry, but do we need subcomponents? Because I need to access some of the parents' state. And since this is not going to be used elsewhere, and the logic was not complex either, so I thought keeping it as a subfunction within the parent component seems like a good approach.

I might be missing something, please let me know if I did.

Reply:
Thank you! I have removed the sub-functions.